tests/hmm/MiscVariables.h

```diff
@@ -44,5 +45,13 @@ extern "C" {
                 out_pt[iev] = PtCorrGeoFit(d0_BS[iev]*(float)(charge[iev]), pt_Roch[iev], eta[iev], year[iev]);
             }
     }
+    void qglJetWeight_eval(
+        float* out_weight, int nev,
+        int* partonFlavour, float* pt, float* eta, float* qgl, int *isHerwig){
+            #pragma omp parallel for default(none) shared(out_weight, nev, partonFlavour, eta, qgl, isHerwig) schedule(dynamic, 1000)
+            for (int iev=0; iev<nev; iev++) {
+                out_weight[iev] = qglJetWeight(partonFlavour[iev], pt[iev], eta[iev], qgl[iev], isHerwig[iev]);
```

Comment:
minor suggestion here: you can pass `isHerwig` as a simple int value, rather than a per-jet array. I guess it's unlikely that in one file you will have both Herwig and non-Herwig events (jets).

Reply:
Thanks a lot for the suggestion. Changed.